lib/openstack_api/openstack_query.py

```diff
@@ -298,3 +304,104 @@ def parse_and_output_table(
+        """
+        Contains the data needed for running find_non_existent_objects
+        :param object_list_func: Function that takes the connection and project and should return all instances
+                                 of the chosen openstack object in that project as a list
+        :param object_get_func: Function that takes the connection and an openstack object id and should attempt
+                                to get that object (Throwing an openstack.exceptions.ResourceNotFound error if it
+                                fails)
+        :param object_id_param_name: Name of the parameter that contains the object id in the objects returned by
+                                     'object_list_func'
+        :param object_project_param_name: Name of the parameter that contains the project id in the objects returned
+                                          by 'object_list_func'
+        """
+
+        object_list_func: Callable[[Connection, Project], List]
+        object_get_func: Callable[[Connection, str], Any]
+        object_id_param_name: str
+        object_project_param_name: str
+
+    def find_non_existent_objects(
+        self,
+        cloud_account: str,
+        project_identifier: str,
+        check_params: NonExistentCheckParams,
+    ) -> Dict[str, List[str]]:
+        """
+        Returns a dictionary containing the ids of projects along with a list of ids of non-existent openstack objects
+        found within them
+        :param cloud_account: The associated clouds.yaml account
+        :param project_identifier: The project to get all associated servers with, can be empty for all projects
+        :param check_params: Parameters required for running the check
+        :return: A dictionary containing the non-existent object ids and their projects
+        """
+        selected_projects = {}
+        if project_identifier == "":
+            projects = self._identity_api.list_projects(cloud_account)
+        else:
+            projects = [
+                self._identity_api.find_mandatory_project(
+                    cloud_account, project_identifier=project_identifier
+                )
+            ]
+
+        with self._connection_cls(cloud_account) as conn:
+            for project in projects:
+                objects_in_project = check_params.object_list_func(conn, project)
+                for obj in objects_in_project:
+                    object_id = obj[check_params.object_id_param_name]
+                    try:
+                        check_params.object_get_func(conn, object_id)
+                    except openstack.exceptions.ResourceNotFound:
+                        if project.id in selected_projects:
+                            selected_projects[project.id].append(object_id)
+                        else:
+                            selected_projects.update({project.id: [object_id]})
+        return selected_projects
+
+    @dataclass
+    class NonExistentProjectCheckParams:
```

Comment:
Can the above dataclass inherit this one to re-use the fields? Since there's duplication going on between these two